core/include/core/G3Reader.h

```diff
@@ -11,10 +11,10 @@ class G3Reader : public G3Module {
 public:
 	G3Reader(std::string filename, int n_frames_to_read = -1,
                  float timeout = -1., bool track_filename = false,
-	         size_t buffersize = 1024*1024);
+	         size_t buffersize = 1024*1024, bool counter = false);
 	G3Reader(std::vector<std::string> filenames, int n_frames_to_read = -1,
                  float timeout = -1., bool track_filename = false,
-	         size_t buffersize = 1024*1024);
+	         size_t buffersize = 1024*1024, bool counter = false);
```

Comment:
Does counter = true cause a measurable performance drop? Or can we just make it also be on?

Reply:
I haven't tried.  Maybe @mhasself or @tskisner can test this out for their use case?

Reply:
Testing actually indicates that this counter introduces a measurable performance drop, so we may need a different approach here.